File: backend/src/main/kotlin/io/github/projektalmanac/comunidades/service/UserService.java
package io.github.projektalmanac.comunidades.service;

import io.github.projektalmanac.comunidades.entity.Comunidad;
import io.github.projektalmanac.comunidades.exception.CommunityNotFoundException;
import io.github.projektalmanac.comunidades.generated.dto.ComunidadDto;
import io.github.projektalmanac.comunidades.generated.dto.IdUsuarioDto;
import io.github.projektalmanac.comunidades.mapper.ComunidadMapper;
import io.github.projektalmanac.comunidades.repository.ComunidadRepository;
import io.github.projektalmanac.comunidades.repository.UserRepository;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class UserService {
    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
    @Autowired
    private UserRepository userRepository;
    @Autowired
    private ComunidadRepository comunidadRepository;

    public ComunidadDto getInfoComunidad(Integer id){
        LOGGER.info(">> getInfoComunidad {}", id);
        Comunidad comunidad = this.comunidadRepository.findComunidadById(id);

        if(comunidad == null){
            throw new CommunityNotFoundException(id);
        }
        ComunidadDto comunidadDto = ComunidadMapper.Companion.getINSTANCE().comunidadToDto(comunidad);
        if(comunidadDto == null){
            throw new IllegalArgumentException("Erro, el mapeo de entidad no fue exitoso");
        }

        LOGGER.info("<< getInfoComunidad {}", id);
        return comunidadDto;

    }

    public void agregaMiembroComunidad(Integer idComunidad, IdUsuarioDto usuarioDto){

    }

}

File: backend/src/main/kotlin/io/github/projektalmanac/comunidades/service/UserService.kt
package io.github.projektalmanac.comunidades.service

import io.github.projektalmanac.comunidades.exception.CommunityNotFoundException
import io.github.projektalmanac.comunidades.generated.dto.ComunidadDto
import io.github.projektalmanac.comunidades.generated.dto.IdUsuarioDto
import io.github.projektalmanac.comunidades.mapper.ComunidadMapper.Companion.INSTANCE
import io.github.projektalmanac.comunidades.repository.ComunidadRepository
import io.github.projektalmanac.comunidades.repository.UserRepository
import org.slf4j.LoggerFactory
import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.jwt.Jwt
import org.springframework.stereotype.Service

@Service
class UserService (private val userRepository: UserRepository, private val comunidadRepository: ComunidadRepository) {
    fun getInfoComunidad(id: Int?): ComunidadDto {
        LOGGER.info(">> getInfoComunidad {}", id)
        val comunidad = comunidadRepository.findComunidadById(id) ?: throw CommunityNotFoundException(id!!)
        val comunidadDto = INSTANCE.comunidadToDto(comunidad)
            ?: throw IllegalArgumentException("Erro, el mapeo de entidad no fue exitoso")
        LOGGER.info("<< getInfoComunidad {}", id)
        return comunidadDto
    }

    fun agregaMiembroComunidad(idComunidad: Int?, usuarioDto: IdUsuarioDto?) {}

    fun getId(authentication: Authentication): Int? {
        val principal = authentication.principal
        require(principal is Jwt)
        val email = principal.claims["email"] as String
        val usuario = userRepository.findByCorreo(email)
        if (usuario === null) return null
        requireNotNull(usuario.idUser)
        return usuario.idUser
    }

    companion object {
        private val LOGGER = LoggerFactory.getLogger(UserService::class.java)
    }
}
